MatrixSDK/Crypto/Verification/MXKeyVerificationManagerV2.swift

```diff
@@ -0,0 +1,236 @@
+//
+//  MXKeyVerificationManagerV2.swift
+//  MatrixSDK
+//
+//  Created by Element on 05/07/2022.
+//
+
+import Foundation
+
+#if DEBUG && os(iOS)
+
+import MatrixSDKCrypto
+
+@available(iOS 13.0.0, *)
+class MXKeyVerificationManagerV2: MXKeyVerificationManager {
+    typealias GetOrCreateDMRoomId = (_ userId: String) async throws -> String
+    
+    override var requestTimeout: TimeInterval {
+        set {
+            log.debug("Not implemented")
+        }
+        get {
+            log.debug("Not implemented")
+            return 1000
+        }
+    }
+    
+    private let verification: MXCryptoVerification
+    private let getOrCreateDMRoomId: GetOrCreateDMRoomId
+    
+    private var requests: [MXKeyVerificationRequestV2]
+    private var transactions: [MXSASTransactionV2]
+    
+    private let log = MXNamedLog(name: "MXKeyVerificationManagerV2")
+    
+    init(verification: MXCryptoVerification, getOrCreateDMRoomId: @escaping GetOrCreateDMRoomId) {
+        self.verification = verification
+        self.getOrCreateDMRoomId = getOrCreateDMRoomId
+        
+        self.requests = []
+        self.transactions = []
+        
+        super.init()
+    }
+    
+    func updatePendingRequests() {
+        for request in requests {
+            guard let req = verification.verificationRequest(userId: request.otherUser, flowId: request.requestId) else {
+                continue
+            }
+            request.update(request: req)
+        }
+        
+        for transaction in transactions {
+            guard let verification = verification.verification(userId: transaction.otherUserId, flowId: transaction.transactionId) else {
+                continue
+            }
```

Comment:
Perhaps it might be useful to log something if we don't get through these guards ?